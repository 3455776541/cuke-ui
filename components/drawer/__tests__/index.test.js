import React from "react";
import { render, shallow, mount } from "enzyme";
import assert from "power-assert";
import toJson from "enzyme-to-json";
import Drawer from "../index";

describe("<Drawer/>", () => {
  it("should render a <Drawer/> components", () => {
    const wrapper = render(
      <div>
        <Drawer title="基本使用" visible>
          <span>基本使用</span>
        </Drawer>
        <Drawer title="基本使用" visible width={500}>
          <span>无 footer</span>
        </Drawer>
        <Drawer title="基本使用" visible width={500} zIndex={9999}>
          <span>无 footer</span>
        </Drawer>
      </div>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it.skip("should can not clicked with set maskClosable false", () => {
    const onCancelClick = jest.fn();
    const wrapper = mount(
      <Drawer visible={true} onCancel={onCancelClick}>
        <span>关闭回调</span>
      </Drawer>
    );
    wrapper.find(".cuke-drawer").simulate("click");
    assert(wrapper.props().visible === true);
    expect(onCancelClick).not.toHaveBeenCalled();
  });

  it("should render custom direction", () => {
    const wrapper = render(
      <div>
        <Drawer visible placement="left">
          <span>关闭回调</span>
        </Drawer>
        <Drawer visible placement="top">
          <span>关闭回调</span>
        </Drawer>
        <Drawer visible placement="right">
          <span>关闭回调</span>
        </Drawer>
        <Drawer visible placement="bottom">
          <span>关闭回调</span>
        </Drawer>
      </div>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should can trigger onClose event", () => {
    const onCancelClick = jest.fn();
    const wrapper = shallow(
      <Drawer title="关闭回调" visible onClose={onCancelClick}>
        <span>关闭回调</span>
      </Drawer>
    );
    wrapper.find(".cuke-drawer").simulate("click");
    expect(onCancelClick).not.toHaveBeenCalled();
  });
});
