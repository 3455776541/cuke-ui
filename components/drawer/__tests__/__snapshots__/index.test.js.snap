// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Drawer/> should render a <Drawer/> components 1`] = `
<div>
  <div
    class="cuke-drawer-mask cuke-drawer-mask-show"
  />
  <div
    class="cuke-drawer-wrap"
    role="dialog"
    tabindex="-1"
  >
    <div
      class="cuke-drawer cuke-drawer-open cuke-drawer-right"
      style="width:300px;z-index:999"
    >
      <section
        class="cuke-drawer-header"
      >
        <h2
          class="cuke-drawer-title"
        >
          基本使用
        </h2>
        <svg
          attr="[object Object]"
          class="undefined cuke-drawer-close"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
        >
          <line
            x1="18"
            x2="6"
            y1="6"
            y2="18"
          />
          <line
            x1="6"
            x2="18"
            y1="6"
            y2="18"
          />
        </svg>
      </section>
      <section
        class="cuke-drawer-content"
      >
        <span>
          基本使用
        </span>
      </section>
    </div>
  </div>
  <div
    class="cuke-drawer-mask cuke-drawer-mask-show"
  />
  <div
    class="cuke-drawer-wrap"
    role="dialog"
    tabindex="-1"
  >
    <div
      class="cuke-drawer cuke-drawer-open cuke-drawer-right"
      style="width:500px;z-index:999"
    >
      <section
        class="cuke-drawer-header"
      >
        <h2
          class="cuke-drawer-title"
        >
          基本使用
        </h2>
        <svg
          attr="[object Object]"
          class="undefined cuke-drawer-close"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
        >
          <line
            x1="18"
            x2="6"
            y1="6"
            y2="18"
          />
          <line
            x1="6"
            x2="18"
            y1="6"
            y2="18"
          />
        </svg>
      </section>
      <section
        class="cuke-drawer-content"
      >
        <span>
          无 footer
        </span>
      </section>
    </div>
  </div>
  <div
    class="cuke-drawer-mask cuke-drawer-mask-show"
  />
  <div
    class="cuke-drawer-wrap"
    role="dialog"
    tabindex="-1"
  >
    <div
      class="cuke-drawer cuke-drawer-open cuke-drawer-right"
      style="width:500px;z-index:9999"
    >
      <section
        class="cuke-drawer-header"
      >
        <h2
          class="cuke-drawer-title"
        >
          基本使用
        </h2>
        <svg
          attr="[object Object]"
          class="undefined cuke-drawer-close"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
        >
          <line
            x1="18"
            x2="6"
            y1="6"
            y2="18"
          />
          <line
            x1="6"
            x2="18"
            y1="6"
            y2="18"
          />
        </svg>
      </section>
      <section
        class="cuke-drawer-content"
      >
        <span>
          无 footer
        </span>
      </section>
    </div>
  </div>
</div>
`;

exports[`<Drawer/> should render custom direction 1`] = `
<div>
  <Drawer
    closable={true}
    maskClosable={true}
    onClose={[Function]}
    placement="left"
    prefixCls="cuke-drawer"
    showMask={true}
    target={[Function]}
    title=""
    visible={true}
    width={300}
    zIndex={999}
  >
    <div
      className="cuke-drawer-mask cuke-drawer-mask-show"
      onClick={[Function]}
    />
    <div
      className="cuke-drawer-wrap"
      role="dialog"
      tabIndex="-1"
    >
      <div
        className="cuke-drawer cuke-drawer-open cuke-drawer-left no-title"
        style={
          Object {
            "width": 300,
            "zIndex": 999,
          }
        }
      >
        <section
          className="cuke-drawer-header"
        >
          <h2
            className="cuke-drawer-title"
          />
          <CloseIcon
            className="cuke-drawer-close"
            onClick={[Function]}
          >
            <Component
              className="cuke-drawer-close"
              onClick={[Function]}
            >
              <IconBase
                attr={
                  Object {
                    "fill": "none",
                    "stroke": "currentColor",
                    "strokeLinecap": "round",
                    "strokeLinejoin": "round",
                    "strokeWidth": "2",
                    "viewBox": "0 0 24 24",
                  }
                }
                className="cuke-drawer-close"
                onClick={[Function]}
              >
                <svg
                  attr={
                    Object {
                      "fill": "none",
                      "stroke": "currentColor",
                      "strokeLinecap": "round",
                      "strokeLinejoin": "round",
                      "strokeWidth": "2",
                      "viewBox": "0 0 24 24",
                    }
                  }
                  className="undefined cuke-drawer-close"
                  fill="none"
                  height="1em"
                  onClick={[Function]}
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 24 24"
                  width="1em"
                >
                  <line
                    key="0"
                    x1="18"
                    x2="6"
                    y1="6"
                    y2="18"
                  />
                  <line
                    key="1"
                    x1="6"
                    x2="18"
                    y1="6"
                    y2="18"
                  />
                </svg>
              </IconBase>
            </Component>
          </CloseIcon>
        </section>
        <section
          className="cuke-drawer-content"
        >
          <span>
            关闭回调
          </span>
        </section>
      </div>
    </div>
  </Drawer>
  <Drawer
    closable={true}
    maskClosable={true}
    onClose={[Function]}
    placement="top"
    prefixCls="cuke-drawer"
    showMask={true}
    target={[Function]}
    title=""
    visible={true}
    width={300}
    zIndex={999}
  >
    <div
      className="cuke-drawer-mask cuke-drawer-mask-show"
      onClick={[Function]}
    />
    <div
      className="cuke-drawer-wrap"
      role="dialog"
      tabIndex="-1"
    >
      <div
        className="cuke-drawer cuke-drawer-open cuke-drawer-top no-title"
        style={
          Object {
            "width": 300,
            "zIndex": 999,
          }
        }
      >
        <section
          className="cuke-drawer-header"
        >
          <h2
            className="cuke-drawer-title"
          />
          <CloseIcon
            className="cuke-drawer-close"
            onClick={[Function]}
          >
            <Component
              className="cuke-drawer-close"
              onClick={[Function]}
            >
              <IconBase
                attr={
                  Object {
                    "fill": "none",
                    "stroke": "currentColor",
                    "strokeLinecap": "round",
                    "strokeLinejoin": "round",
                    "strokeWidth": "2",
                    "viewBox": "0 0 24 24",
                  }
                }
                className="cuke-drawer-close"
                onClick={[Function]}
              >
                <svg
                  attr={
                    Object {
                      "fill": "none",
                      "stroke": "currentColor",
                      "strokeLinecap": "round",
                      "strokeLinejoin": "round",
                      "strokeWidth": "2",
                      "viewBox": "0 0 24 24",
                    }
                  }
                  className="undefined cuke-drawer-close"
                  fill="none"
                  height="1em"
                  onClick={[Function]}
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 24 24"
                  width="1em"
                >
                  <line
                    key="0"
                    x1="18"
                    x2="6"
                    y1="6"
                    y2="18"
                  />
                  <line
                    key="1"
                    x1="6"
                    x2="18"
                    y1="6"
                    y2="18"
                  />
                </svg>
              </IconBase>
            </Component>
          </CloseIcon>
        </section>
        <section
          className="cuke-drawer-content"
        >
          <span>
            关闭回调
          </span>
        </section>
      </div>
    </div>
  </Drawer>
  <Drawer
    closable={true}
    maskClosable={true}
    onClose={[Function]}
    placement="right"
    prefixCls="cuke-drawer"
    showMask={true}
    target={[Function]}
    title=""
    visible={true}
    width={300}
    zIndex={999}
  >
    <div
      className="cuke-drawer-mask cuke-drawer-mask-show"
      onClick={[Function]}
    />
    <div
      className="cuke-drawer-wrap"
      role="dialog"
      tabIndex="-1"
    >
      <div
        className="cuke-drawer cuke-drawer-open cuke-drawer-right no-title"
        style={
          Object {
            "width": 300,
            "zIndex": 999,
          }
        }
      >
        <section
          className="cuke-drawer-header"
        >
          <h2
            className="cuke-drawer-title"
          />
          <CloseIcon
            className="cuke-drawer-close"
            onClick={[Function]}
          >
            <Component
              className="cuke-drawer-close"
              onClick={[Function]}
            >
              <IconBase
                attr={
                  Object {
                    "fill": "none",
                    "stroke": "currentColor",
                    "strokeLinecap": "round",
                    "strokeLinejoin": "round",
                    "strokeWidth": "2",
                    "viewBox": "0 0 24 24",
                  }
                }
                className="cuke-drawer-close"
                onClick={[Function]}
              >
                <svg
                  attr={
                    Object {
                      "fill": "none",
                      "stroke": "currentColor",
                      "strokeLinecap": "round",
                      "strokeLinejoin": "round",
                      "strokeWidth": "2",
                      "viewBox": "0 0 24 24",
                    }
                  }
                  className="undefined cuke-drawer-close"
                  fill="none"
                  height="1em"
                  onClick={[Function]}
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 24 24"
                  width="1em"
                >
                  <line
                    key="0"
                    x1="18"
                    x2="6"
                    y1="6"
                    y2="18"
                  />
                  <line
                    key="1"
                    x1="6"
                    x2="18"
                    y1="6"
                    y2="18"
                  />
                </svg>
              </IconBase>
            </Component>
          </CloseIcon>
        </section>
        <section
          className="cuke-drawer-content"
        >
          <span>
            关闭回调
          </span>
        </section>
      </div>
    </div>
  </Drawer>
  <Drawer
    closable={true}
    maskClosable={true}
    onClose={[Function]}
    placement="bottom"
    prefixCls="cuke-drawer"
    showMask={true}
    target={[Function]}
    title=""
    visible={true}
    width={300}
    zIndex={999}
  >
    <div
      className="cuke-drawer-mask cuke-drawer-mask-show"
      onClick={[Function]}
    />
    <div
      className="cuke-drawer-wrap"
      role="dialog"
      tabIndex="-1"
    >
      <div
        className="cuke-drawer cuke-drawer-open cuke-drawer-bottom no-title"
        style={
          Object {
            "width": 300,
            "zIndex": 999,
          }
        }
      >
        <section
          className="cuke-drawer-header"
        >
          <h2
            className="cuke-drawer-title"
          />
          <CloseIcon
            className="cuke-drawer-close"
            onClick={[Function]}
          >
            <Component
              className="cuke-drawer-close"
              onClick={[Function]}
            >
              <IconBase
                attr={
                  Object {
                    "fill": "none",
                    "stroke": "currentColor",
                    "strokeLinecap": "round",
                    "strokeLinejoin": "round",
                    "strokeWidth": "2",
                    "viewBox": "0 0 24 24",
                  }
                }
                className="cuke-drawer-close"
                onClick={[Function]}
              >
                <svg
                  attr={
                    Object {
                      "fill": "none",
                      "stroke": "currentColor",
                      "strokeLinecap": "round",
                      "strokeLinejoin": "round",
                      "strokeWidth": "2",
                      "viewBox": "0 0 24 24",
                    }
                  }
                  className="undefined cuke-drawer-close"
                  fill="none"
                  height="1em"
                  onClick={[Function]}
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 24 24"
                  width="1em"
                >
                  <line
                    key="0"
                    x1="18"
                    x2="6"
                    y1="6"
                    y2="18"
                  />
                  <line
                    key="1"
                    x1="6"
                    x2="18"
                    y1="6"
                    y2="18"
                  />
                </svg>
              </IconBase>
            </Component>
          </CloseIcon>
        </section>
        <section
          className="cuke-drawer-content"
        >
          <span>
            关闭回调
          </span>
        </section>
      </div>
    </div>
  </Drawer>
</div>
`;
